apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/RowEditor/RowEditor.utils.ts

```diff
@@ -36,17 +37,26 @@ export const generateRowFields = (
                 ? convertPostgresDatetimeToInputDatetime(column.format, row[column.name])
                 : parseValue(row[column.name], column.format)
 
-    const foreignKey = find(relationships, (relationship) => {
-      return (
-        relationship.source_schema === column.schema &&
-        relationship.source_table_name === column.table &&
-        relationship.source_column_name === column.name
-      )
+    const foreignKey = foreignKeys.find((fk) => {
+      return fk.columns.map((x) => x.source).includes(column.name)
     })
```

Comment:
Just to check, checking the `schema` is not important here anymore?

Reply:
actually yeah good point - should add that check just to be sure as there could be same column names across schemas

Reply:
oh yeap this is okay - cause we already filter the foreign keys one level above in RowEditor
https://github.com/supabase/supabase/blob/e2f09af1009da3b355054da915355f2ae273b614/apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/RowEditor/RowEditor.tsx#L67